### ISSUE TITLE
destroy function, imageswap

### ISSUE THREAD

**Issue by maetz3** [imgix/drift#19]
Setup:
```
<a data-zoom="image1.jpg" data-trigger="0"><img></a>

<img data-src="Image2.jpg" data-trigger="1">
<img data-src="Image3.jpg" data-trigger="0">
<img data-src="Image4.jpg" data-trigger="1">
<img data-src="Image5.jpg" data-trigger="0">
```

images with data-trigger on click 
replace values
`<a data-zoom="this image src" data-trigger="this value"><img src etc></a>`

in this case: Image2 + 4 trigger = 1

js:
```
var driftvar;
if data-trigger = 1 driftvar = new Drift...
else 
driftvar.destroy()

```
scenario: 
On click image2 and this trigger value is 1, Drift works perfect.
On click Image3 and trigger value is 0, Drift destroy works perfect.

But:
click image2 then
click image4 then 
click image3 with trigger = 0, 
Drift destory no longer works.

destroy only works if, trigger 1 0 1 0 1 0

**Comment by paulstraw** [comment]
Hey @maetz3, I'm not totally sure I follow what all is happening here. Could you possibly set up a CodePen to showcase this issue?

**Comment by maetz3** [comment]
Hey,

hope this help (a basic function)
http://jsbin.com/dihasibaza/

if only one image present, destroy works perfect (resize, etc).
my thought was : if a picture is less than 800px then remove zoom otherwise zoom.

in the jsbin case: Drift stop at 0 but if 1, Drift append and append etc etc

my current solution , if data-trigger 1 .length then zoom all images with 0 and 1.
if data-trigger 1 not found, no zoom. 

`var morePics = $('.images a');

$(morePics).click(function(e){
e.preventDefault();
var zoomtrigger = $(this).find('img').attr('data-trigger'),
zoomswitch = $(this).attr('data-zoom');

$(this).closest('#container').find('#defaultimage img').attr('data-trigger', zoomtrigger);		
$(this).closest('#container').find('#defaultimage img').attr('src', zoomswitch);
$(this).closest('#container').find('#defaultimage a').attr('data-zoom', zoomswitch);
$(this).trigger('pushthis')
});
$(document).on('pushthis', function() {
Trigger = document.querySelector('.mainimage');
Container = document.querySelector('#zoomcontainer');
var zoomtrigger = $('.mainimage img').attr('data-trigger'),
zoomimg;
if(zoomtrigger === "1") {
zoomimg = new Drift(Trigger, {
paneContainer: Container,
inlinePane: false,
showWhitespaceAtEdges: true,
zoomFactor: 2,
});
} else {
zoomimg.destroy();
}
});
`

**Comment by paulstraw** [comment]
Hey @maetz3, After looking through this example, I'm still not totally sure what your goal is here. Can you try to rephrase it, possibly?

**Comment by maetz3** [comment]
Hi,
i try to disable zoom on images, if they < 600x600
If > 600 trigger 1 else  trigger 0
On click event, Drift destory and reinit if trigger 1.

Work: If one Image, Init Switch works. Or if click Trigger 1 0 1 0
Not Work: If more Images and Trigger 0 and 1, destroy fails after Trigger 1 1 0 click
also tested: destroy and then null zoomimg (works on resize with reinit)
DOM zoomimg = null but zoom shows up on hover.
DOM Drift e(t)

On resize event Drift destory and reinit with inline pane works but only with custom destroy like: 
`if ($(zoomActivate).length) {
 zoomimg.destroy();
} else {
zoomimg = null;
}
zoomActivate = null;
Trigger = null;
Container = null;
morePics = null;`

otherwise Drift show inline pane and zoom window :-)

btw IE9 return classList error in injectBaseStyles function

i think i try a new solution with reinit Drift and hidden zoom Pane DIV or zoomfactor 0

many thanks

**Comment by cezr** [comment]
Hey @maetz3, I just started trying out this zoom library. Anyhow, if I read your issue correctly your having issues with: 1) how to properly destroy the drift object and 2) work with multiple images and 3) how to init and destroy. 

Either way I had an issue relatively similar to yours where I had multiple preview like small images with different src images and data-zoom images. On each click of a preview image then create a new
drift object to load a new in a new zoom. I ran into a issue like yours where it works the 2 times then just stops.

A solution that worked for me is like this example.
var $drift = null;
function init_event_handlers() {
$drift = new Drift(document.querySelector('#zoom_image'),{
        zoomFactor: 2,
        inlinePane: 200,
        inlineOffsetX: 0,
        inlineOffsetY: 0,
        paneContainer: document.querySelector('.big-image-wrap'),
        containInline: true,
        // hoverBoundingBox: true
      });

$(".show-image-preview").click(show_image);
}

function show_image(evt)  {
.... // jquery stuff
...
if ($drift != null) 
{
        $drift.destroy();
        $drift = null;

        setTimeout(function(){
          console.log("init new drift");
          $drift = new Drift(document.querySelector('#zoom_image'),{
            zoomFactor: 2,
            inlinePane: 200,
            inlineOffsetX: 0,
            inlineOffsetY: 0,
            paneContainer: document.querySelector('.big-image-wrap'),
            containInline: true,
            // hoverBoundingBox: true
          });
        }, 10);
}
init_event_handlers();

Solution observation:
Just wrap the new drift object in a setTimeout async function with delay 0 or 100. I think the idea here is to allow the dom to async update then a new Drift object will work.
